### PR TITLE
Scene importer: add -rigidonly flag for empties (fixes #8430)

### LINKED ISSUE #8430: Feature request: -rigidonly
Among other things Godot's scene importer currently supports:

* a `-col` tag that when importing a mesh will also make an accompanying `StaticBody` based on it
* a `-colonly` tag that imports a mesh only as a `StaticBody`, with no visual mesh, and additionally supports Blender's Empty objects for defining collision shapes like spheres, boxes and so on
* a `-rigid` tag that is like the `-col` tag, but makes a `RigidBody`

However, there is no `-rigidonly` counterpart to `-colonly` for `RigidBody` collisions, which wouldn't produce a visual mesh, and would support Blender's Empty shapes.

It would be nice if there was a `-rigidonly` tag. In the meantime, I'm using `-colonly` on my sphere-shaped Blender Empty along with a small post-processing script:

```GDscript
tool
extends EditorScenePostImport

func post_import(scene):
	var sphere = scene.get_node('SphereCollider')
	var newSphere = RigidBody.new()
	newSphere.set_transform(sphere.get_transform())
	sphere.replace_by(newSphere, true)
	return scene
```

### PR DESCRIPTION
**Pull request by hikari-no-yume**

### CODE REVIEW

**Comment by hikari-no-yume** [comment]
Willing to rebase into a single commit if desired.

I haven't actually tested the second commit, because it's a pain to test this. I can't see why it wouldn't work, though.

File: editor/io_plugins/editor_scene_import_plugin.cpp [hunk 1]
```diff
@@ -1567,7 +1567,9 @@ Node *EditorSceneImportPlugin::_fix_node(Node *p_node, Node *p_root, Map<Ref<Mes
 		mi->set_baked_light_texture_id(layer);
 	}
 
-	if (p_flags & SCENE_FLAG_CREATE_COLLISIONS && _teststr(name, "colonly")) {
+	bool is_rigid = false;
+
+	if (p_flags & SCENE_FLAG_CREATE_COLLISIONS && (_teststr(name, "colonly") || (is_rigid = _teststr(name, "rigidonly") && !p_node->cast_to<MeshInstance>()))) {
```
Comment (akien-mga) [inline editor/io_plugins/editor_scene_import_plugin.cpp]: Please don't do assignments in a conditional, that's bad practice and makes the code hard to read.

**Comment by akien-mga** [comment]
If you can rebase into a single commit, that would be nice yes. Also make a PR for the master branch, which should stay the main development branch. I don't want features in 2.1 that would be absent from master.

File: editor/io_plugins/editor_scene_import_plugin.cpp (hunk 1 above)
Comment (hikari-no-yume) [inline editor/io_plugins/editor_scene_import_plugin.cpp]: That's fair, they're not always the best choice. How would you prefer to structure this code?

**Comment by hikari-no-yume** [comment]
Ah, Godot doesn't merge from 2.1 upwards into master?

File: editor/io_plugins/editor_scene_import_plugin.cpp (hunk 1 above)
Comment (hikari-no-yume) [inline editor/io_plugins/editor_scene_import_plugin.cpp]: Would doing `bool is_rigid = _teststr(name, "rigidonly");`, then testing against `is_rigid` be better?

**Comment by Zireael07** [comment]
@hikari-no-yume: My understanding is that the two branches are wildly divergent in some aspects so you can't just merge from one to the other.

**Comment by hikari-no-yume** [comment]
@Zireael07: Huh, okay.

**Comment by hikari-no-yume** [comment]
Rebased against current 2.1 and into a single commit, and moved assignment out of the conditional, as per @akien-mga's suggestion.

**Comment by hikari-no-yume** [comment]
@akien-mga PR for master here: https://github.com/godotengine/godot/pull/8592

**Pull request merged**